lib/vector-config-common/src/human_friendly.rs

```diff
@@ -0,0 +1,108 @@
+use std::collections::HashMap;
+
+use convert_case::{Boundary, Case, Casing};
+use once_cell::sync::OnceCell;
+
+static WELL_KNOWN_REPLACEMENTS: OnceCell<HashMap<String, &'static str>> = OnceCell::new();
+
+/// Methods for splitting an input string into word segments.
+pub enum SplitMethod {
+    /// Split inputs on case change boundaries.
+    Case,
+
+    /// Split inputs on underscore boundaries.
+    Underscore,
+}
```

Comment:
I had that thought that `convert_case` might do that, but hadn't explored it. I figured that depending on the usage, we'd always know the input format: enum variants are always PascalCase, and structs are always snake_case, and so specifying the split method manually at the individual callsites was good enough.

But it would be nice to make this simpler, so I can take a looksie.